### PR TITLE
Replaced switch with a regexp

### PR DESCRIPTION
**Pull request by jmisiti42**

### CODE REVIEW

**Comment by ry** [comment]
Hi - thanks for the submission - but given there's no change in behavior I think this needs to be justified with a benchmark... Note this is called for every request, so it's very hot.

File: std/http/server.ts [hunk 1]
```diff
@@ -234,53 +234,22 @@ function fixLength(req: ServerRequest): void {
  * Ported from https://github.com/golang/go/blob/f5c43b9/src/net/http/request.go#L766-L792
  */
 export function parseHTTPVersion(vers: string): [number, number] {
-  switch (vers) {
-    case "HTTP/1.1":
-      return [1, 1];
+  const httpReg = /^HTTP\/([0-9]+)\.([0-9]+)$/gm;
```
Comment (nayeemrmn) [inline std/http/server.ts]: What's the `gm` for?

Comment (jmisiti42) [reply std/http/server.ts]: You'r right, this `gm` is useless.

**Comment by jmisiti42** [comment]
No need to review this as it will be refactored

**Comment by ry** [comment]
Closing. Re-open if justified.

**Pull request closed without merging**